### PR TITLE
Moved to RMSE for image comparison to account for slight variations in golden image tests

### PR DESCRIPTION
**Pull request by gaaclarke**
## Description

There appears to be slight variations in golden images between different people's machines despite running on the same simulator os version.  This will run a tight comparison to make sure the root-square error is less than 0.5.

## Related Issues

https://github.com/flutter/flutter/issues/56388

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #import <XCTest/XCTest.h>
 #include <sys/sysctl.h>
 
+static const double kRmseThreshold = 0.5;
```
Comment (blasten) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:9]: how did you determinate this value?

Comment (gaaclarke) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:9]: I ran the tests on my machine without changes and that was the smallest threshold that that contains all the errors, plus it is a reasonably low error.

**Comment by gaaclarke** [comment]
@blasten I started ignoring transparent pixels like we discussed offline Friday.  I also had to tweak one of the assertions you added to your test which was flakey for jenn and I locally.  Can you please check that out to make sure the test is still good.  I think you just wanted to assert that you have 1 overlay which is inside of the view.  You don't really care what the exact size is.

**Comment by gaaclarke** [comment]
@jmagman This should also fix the flake we were both seeing, I doubt it will do anything for https://github.com/flutter/flutter/issues/61267 but maybe.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m (hunk 1 above)
Comment (blasten) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:9]: what would be an example of a false positive? 

If you change https://github.com/flutter/engine/blob/master/testing/scenario_app/lib/src/platform_view.dart#L523 to `151` or one of the translates by `1` pixel?

Comment (blasten) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:9]: what would be an example of a false positive?

If you change https://github.com/flutter/engine/blob/master/testing/scenario_app/lib/src/platform_view.dart#L523 to 151 or one of the translates by 1 pixel, does the test pass?

Comment (gaaclarke) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:9]: It doesn't pass when i change it to 151.  In that case the RMSE goes from 0.49 to 0.94.

Comment (gaaclarke) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:9]: RMSE factors in how big of a diff there is and is relative to the size of the images.  So small discrepancies on large images make less of a difference.

Comment (blasten) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m:9]: SGTM

**Pull request merged**